### PR TITLE
Update docker selection guidence in the UX #2518

### PR DESCRIPTION
**Pull request by yxwithu**
https://github.com/Microsoft/pai/issues/2518

### CODE REVIEW

File: contrib/submit-nfs-job/src/App/SimpleNFSJob.tsx [hunk 1]
```diff
@@ -43,13 +45,16 @@ interface ISimpleNFSJobObject {
 export default class SimpleNFSJob extends Job {
   public constructor(
     private readonly name: string,
+    private readonly image: string,
+    private readonly vc: string,
```
Comment (Gerhut) [inline contrib/submit-nfs-job/src/App/SimpleNFSJob.tsx]: Use "virtualCluster" instead

File: contrib/submit-nfs-job/src/App/index.tsx [hunk 2]
```diff
@@ -133,26 +135,38 @@ export default function App({ pluginId, api, user, token, originalJobName, origi
...
     setName(`${type}-${Date.now().toString(36).toLocaleUpperCase()}`);
+    switch(type){
+      case "single-task": 
+        setImage("openpai/pai.build.base:hadoop2.7.2-cuda9.0-cudnn7-devel-ubuntu16.04");
+        break;
+      case "tensorflow-single-task": 
+      case "tensorflow-distributed":
+        setImage("openpai/pai.example.tensorflow:stable");
+        break;
+      default: break;
```
Comment (Gerhut) [inline contrib/submit-nfs-job/src/App/index.tsx]: Does this line just for linting?

File: contrib/submit-nfs-job/src/App/TensorflowDistributedJob.tsx [hunk 3]
```diff
@@ -154,6 +164,8 @@ echo "All workers are: $PAI_TASK_ROLE_worker_HOST_LIST"
   useEffect(() => {
     onChange(new TensorflowDistributedJob(
       name,
+      image,
+      vc,
```
Comment (Gerhut) [inline contrib/submit-nfs-job/src/App/TensorflowDistributedJob.tsx]: please also add these 2 field as dependencies of `useEffect`

Comment (yxwithu) [reply contrib/submit-nfs-job/src/App/TensorflowDistributedJob.tsx]: thanks, I updated the code

File: contrib/submit-nfs-job/src/App/index.tsx [hunk 4]
```diff
@@ -58,8 +58,10 @@ interface IProps {
 
 export default function App({ pluginId, api, user, token, originalJobName, originalJobUser }: IProps) {
   const [job, setJob] = useState<Job | null>(null);
-  const [type, onTypeChanged, setType] = useValue("simple-nfs");
+  const [type, onTypeChanged, setType] = useValue("single-task");
   const [name, onNameChanged, setName] = useValue(`unnamed-job-${Date.now()}`);
+  const [image, onImageChanged, setImage] = useValue("openpai/pai.build.base:hadoop2.7.2-cuda9.0-cudnn7-devel-ubuntu16.04"); 
+  const [virtualCluster, onVCChanged, setVC] = useValue("default");
```
Comment (Gerhut) [inline contrib/submit-nfs-job/src/App/index.tsx]:
```suggestion
  const [virtualCluster, onVirtualClusterChanged, setVirtualCluster] = useValue("default"); 
```

File: contrib/submit-nfs-job/src/App/index.tsx [hunk 5]
```diff
@@ -133,26 +135,37 @@ export default function App({ pluginId, api, user, token, originalJobName, origi
   useEffect(() => {
     if (originalJob != null) {
       setType(originalJob.type);
+      setImage(originalJob.image);
+      setVC(originalJob.virtualCluster);
```
Comment (Gerhut) [inline contrib/submit-nfs-job/src/App/index.tsx]:
```suggestion
      setVirtualCluster(originalJob.virtualCluster);
```

File: contrib/submit-nfs-job/src/App/index.tsx [hunk 6]
```diff
@@ -182,6 +195,38 @@ export default function App({ pluginId, api, user, token, originalJobName, origi
...
+            <div className="form-group">
+              <label htmlFor="virtual-cluster">Virtual Cluster</label>
+              <div className="row">
+                <div className="col-sm-5">
+                  <input
+                    type="text"
+                    className="form-control"
+                    id="virtual-cluster"
+                    value={virtualCluster}
+                    onChange={onVCChanged}
```
Comment (Gerhut) [inline contrib/submit-nfs-job/src/App/index.tsx]:
```suggestion
                    onChange={onVirtualClusterChanged}
```

File: contrib/submit-nfs-job/src/App/TensorflowSingleNodeJob.tsx [hunk 7]
```diff
@@ -110,7 +118,7 @@ python train_image_classifier.py --batch_size=64 --model_name=inception_v3 --dat
   const [mountDirectories, setMountDirectories] = useState<MountDirectories | null>(null);
 
   useEffect(() => {
-    onChange(new TensorflowSingleNodeJob(name, gpuNumber, command, mountDirectories));
+    onChange(new TensorflowSingleNodeJob(name, image, virtualCluster, gpuNumber, command, mountDirectories));
   }, [name, gpuNumber, command, mountDirectories]);
```
Comment (Gerhut) [inline contrib/submit-nfs-job/src/App/TensorflowSingleNodeJob.tsx]:
```suggestion
   }, [name, image, virtualCluster, gpuNumber, command, mountDirectories]);
```

**Comment by Gerhut** [comment]
Please fix the lint warnings

**Comment by Gerhut** [comment]
:shipit:

**Pull request merged**